### ISSUE TITLE
Document current "Create a release" process

### ISSUE THREAD

**Issue by Potherca** [Dealerdirect/phpcodesniffer-composer-installer#114]
In order to guarantee consistencies with releases from this codebase, the process used to create a new release needs to be documented.

Roughly, it goes like this:

1.Make sure all closed tickets and MRs have a label.
2. Move any open tickets to the next milestone (create a new one if needed)
3. Run [github-changelog-generator](https://github.com/github-changelog-generator/github-changelog-generator) to generate a changelog<sup>1</sup>
4. Use GitHub "Draft a new release" functionality to draft a new release (this also creates a tag).
5. Close the milestone for the version that was just released

I'll write things down in more detail.

Maybe later we can also have a look at automating this step with a GitHub Action (i.e. push-button release).

<sup>1</sup> I currently user these setting: [potherca-contrib/phpcodesniffer-composer-installer/.github_changelog_generator](https://github.com/potherca-contrib/phpcodesniffer-composer-installer/blob/master/.github_changelog_generator)
And this script: [potherca-contrib/phpcodesniffer-composer-installer/bin/generate-changelog.sh](https://github.com/potherca-contrib/phpcodesniffer-composer-installer/blob/generate-changelog/bin/generate-changelog.sh)

**Comment by jrfnl** [comment]
Might be helpful for the changelog generation tooling (settings, script) to be added to this repo.

I imagine it may also be useful for the above steps to be included in the `CONTRIBUTING.md` doc ?

**Comment by Potherca** [comment]
Agreed, on both points.

**Comment by jrfnl** [comment]
While looking at the Changelog for the `1.7.1` release I noticed two bugs:
1. Underscores being escaped - `installed\_paths`.
2. An issue from another milestone having been added to the changelog - #108 was fixed in `1.7.0` not `1.7.1`.

I've manually edited the release now to fix both, but I think it would be good to review the configuration of the changelog generator to prevent these issues in the future.

Screenshot of the original changelog/release announcement:
![image](https://user-images.githubusercontent.com/663378/101407750-a3152f80-38db-11eb-876f-91ae84b34b75.png)
